src/typedefs.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable max-len */
+// typedefs.js
+
+/**
+ * @typedef {Object} EventHandlers
+ * @property {function} onReady - Called when HubSpot is ready to receive messages.
+ * @property {function} onDialNumber - Called when the HubSpot sends a dial number from the contact.
+ * @property {function} onEngagementCreated - Called when HubSpot creates an engagement
+ * for the call.
+ * @property {function} onVisibilityChanged - Called when the call widget's visibility changes.
+ * @property {function} [defaultEventHandler] - Default event handler to handle unhandled events.
+ */
+
+/**
+ * @typedef {Object} IframeOptions
+ * @property {string} src - iframe URL
+ * @property {string} height - Height of iframe
+ * @property {string} width - Width of iframe
+ * @property {string} hostElementSelector - Selector for host element where iframe will be bound
+ */
+
+/**
+ * @typedef {Object} Options
+ * @property {IframeOptions} iFrameOptions - iFrame configuration options
+ * @property {boolean} debugMode - Whether to log various inbound/outbound debug messages
+ * to the console.
+ * @property {EventHandlers} eventHandlers - Event handlers handle inbound messages.
+ */
+
+/**
+ * @typedef {Object} OnResize
+ * @property {number} width
+ * @property {number} height
+ */
+
+/**
+ * @typedef {Object} OnInitialized
+ * @property {boolean} [isLoggedIn]
+ * @property {number} [engagementId]
+ * @property {OnResize} [sizeInfo] - @deprecated
+ */
+
+/**
+ * @typedef {Object} OnIncomingCall
+ * @property {number} externalCallId
+ * @property {number} [callStartTime]
+ * @property {boolean} [createEngagement]
+ * @property {string} fromNumber
+ * @property {string} toNumber
+ */
+
+/**
+ * @typedef {Object} OnOutgoingCall
+ * @property {number} externalCallId
+ * @property {number} [callStartTime]
+ * @property {boolean} [createEngagement]
+ * @property {string} [fromNumber]
+ * @property {string} [toNumber]
+ * @property {string} phoneNumber - @deprecated use toNumber instead
+ */
+
+/**
+ * @typedef {Object} OnCallAnswered
+ * @property {number} externalCallId
+ */
+
+/**
+ * @typedef {Object} OnPublishToChannel
+ * @property {number} externalCallId
```

Comment:
```suggestion
 * @property {number} [externalCallId]
```

Reply:
It is required in Payload.ts on sdk-support side

Reply:
Sorry, I flipped them. I thought [] made them required